src/main/java/org/terasology/caves/CaveFacetProvider.java

```diff
@@ -35,233 +36,36 @@
 @RegisterPlugin
 @Produces(CaveFacet.class)
 @Requires(@Facet(SurfaceHeightFacet.class))
-public class CaveFacetProvider implements ConfigurableFacetProvider, FacetProviderPlugin {
-    private CaveFacetProviderConfiguration configuration = new CaveFacetProviderConfiguration();
+public class CaveFacetProvider implements FacetProviderPlugin {
 
-    Noise baseCaveNoise;
-    Noise baseFadeCaveNoise;
+    SubSampledNoise3D[] caveNoise = new SubSampledNoise3D[2];
 
     @Override
     public void setSeed(long seed) {
-        baseCaveNoise = new PerlinNoise(seed + 2);
-        baseFadeCaveNoise = new PerlinNoise(seed + 3);
+        for(int i=0; i<2; i++) {
+            Noise3D pNoise = new SimplexNoise(seed + 2 + i);
+            BrownianNoise3D bNoise = new BrownianNoise3D(pNoise, 3);
+            bNoise.setPersistence(1.0);
+            SubSampledNoise3D sNoise = new SubSampledNoise3D(bNoise, new Vector3f(0.01f, 0.01f, 0.01f), 4);
+            caveNoise[i] = sNoise;
+        }
     }
 
     @Override
     public void process(GeneratingRegion region) {
-        float width = configuration.width;
-        float height = configuration.height;
-        float gradualIncreaseOverDepth = configuration.gradualIncreaseOverDepth;
-        float amountOfCavesNearSurface = configuration.amountofCavesNearSurface;
-        float sharpSurfaceCutoffDepth = 12;
-        float minDepth = 0;
-        float amountOfCaves = configuration.amountOfCaves;
-        float noiseLevel = configuration.rawAmount;
 
-        // at default settings,  make caves wider than tall.
-        SubSampledNoise caveNoise = new SubSampledNoise(new RidgedNoise(baseCaveNoise, 2), new Vector3f(0.06f * (1f / width), 0.09f * (1f / height), 0.06f * (1f / width)), 4);
-        SubSampledNoise fadeCaveNoise = new SubSampledNoise(baseFadeCaveNoise, new Vector3f(0.006f * (1f / width), 0.006f * (1f / height), 0.006f * (1f / width)), 1);
         CaveFacet facet = new CaveFacet(region.getRegion(), region.getBorderForFacet(CaveFacet.class));
-        SurfaceHeightFacet surfaceHeightFacet = region.getRegionFacet(SurfaceHeightFacet.class);
 
         // get noise in batch for performance reasons.  Getting it by individual position takes 10 times as long
```

Comment:
We should put a note on that in our world gen tutorial if it is that critical o.O

Reply:
Put a note about what? I don't understand.

Reply:
I think iterating over the facet points is a bit clearer to read/more intuitive, but if the performance impact is an order of magnitude we should explain the _batch_ approach (using arrays) in the corresponding tutorial.

Not strictly related to this change, just got aware of this...